cmd/dra-example-controller/driver.go

```diff
@@ -133,17 +141,86 @@ func (d driver) Allocate(ctx context.Context, claim *resourcev1.ResourceClaim, c
 		err = fmt.Errorf("unknown ResourceClaim.ParametersRef.Kind: %v", claim.Spec.ParametersRef.Kind)
 	}
 	if err != nil {
-		return nil, fmt.Errorf("unable to allocate devices on node '%v': %v", selectedNode, err)
+		return fmt.Errorf("unable to allocate devices on node '%v': %v", selectedNode, err)
 	}
 
 	err = client.Update(&crd.Spec)
 	if err != nil {
-		return nil, fmt.Errorf("error updating NodeAllocationState CRD: %v", err)
+		return fmt.Errorf("error updating NodeAllocationState CRD: %v", err)
 	}
 
 	onSuccess()
 
-	return buildAllocationResult(selectedNode, true), nil
+	return nil
+}
+
+func (d driver) allocateImmediateClaim(claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}) (*resourcev1.AllocationResult, error) {
+
+	crdconfig := &nascrd.NodeAllocationStateConfig{
+		Namespace: d.namespace,
+	}
+	crd := nascrd.NewNodeAllocationState(crdconfig)
+
+	client := nasclient.New(crd, d.clientset.NasV1alpha1())
```

Comment:
This can be pulled up into `Allocate()` with the client passed directly to the `allocateDelayedClaim()` and `allocateImmediateClaim()` calls.